cluster-autoscaler/cloudprovider/azure/azure_scale_set.go

```diff
@@ -447,21 +531,12 @@ func (scaleSet *ScaleSet) DeleteInstances(instances []*azureRef, hasUnregistered
 
 	ctx, cancel := getContextWithTimeout(vmssContextTimeout)
 	defer cancel()
-	resourceGroup := scaleSet.manager.config.ResourceGroup
 
-	scaleSet.instanceMutex.Lock()
-	klog.V(3).Infof("Calling virtualMachineScaleSetsClient.DeleteInstancesAsync(%v), force delete set to %v", requiredIds.InstanceIds, scaleSet.enableForceDelete)
-	future, rerr := scaleSet.manager.azClient.virtualMachineScaleSetsClient.DeleteInstancesAsync(ctx, resourceGroup, commonAsg.Id(), *requiredIds, scaleSet.enableForceDelete)
+	future, rerr := scaleSet.deleteInstances(ctx, requiredIds, commonAsg.Id())
 
-	if scaleSet.enableForceDelete && isOperationNotAllowed(rerr) {
-		klog.Infof("falling back to normal delete for instances %v for %s", requiredIds.InstanceIds, scaleSet.Name)
-		future, rerr = scaleSet.manager.azClient.virtualMachineScaleSetsClient.DeleteInstancesAsync(ctx, resourceGroup,
-			commonAsg.Id(), *requiredIds, false)
-	}
-
-	scaleSet.instanceMutex.Unlock()
 	if rerr != nil {
-		klog.Errorf("virtualMachineScaleSetsClient.DeleteInstancesAsync for instances %v failed: %v", requiredIds.InstanceIds, rerr)
```

Comment:
Not present on older versions due to fork 1.30 have this white space for unknown reason... No commit differences. Probably a merge mistake.